TAGGING.md

```diff
@@ -0,0 +1,86 @@
+It is possible you are using objectfs with multiple environments (e.g. prod, staging) that both point to the same bucket. Since files are referenced by contenthash, it generally does not matter where they come from, so this isn't a problem. However to ensure the tags remain accurate, you should turn off `overwriteobjecttags` in the plugin settings for every environment except production.
+
+This means that staging is unable to overwrite tags for files uploaded elsewhere, but can set it on files only uploaded only from staging. However, files uploaded from production will always have the correct tags, and will overwrite any existing tags. 
+
+```mermaid
+graph LR
+    subgraph S3
+        Object("`**Object**
+        contenthash: xyz
+        tags: env=prod`")
+    end
+    subgraph Prod
+        UploadObjectProd["`**Upload object**
+        contenthash: xyz
+        tags: env=prod`"] --> Object
+    end
+    subgraph Staging
+        UploadObjectStaging["`**Upload object**
+        contenthash: xyz
+        tags: env=staging`"]
+    end
+    Blocked["Blocked - does not have permissions\nto overwrite existing object tags"]
+    UploadObjectStaging --- Blocked
+    Blocked -.-> Object
+
+    style Object fill:#ffffff00,stroke:#ffa812
+    style S3 fill:#ffffff00,stroke:#ffa812
+    style Prod fill:#ffffff00,stroke:#26ff4a
+    style UploadObjectProd fill:#ffffff00,stroke:#26ff4a
+    style Staging fill:#ffffff00,stroke:#978aff
+    style UploadObjectStaging fill:#ffffff00,stroke:#978aff
+    style Blocked fill:#ffffff00,stroke:#ff0000
+```
+
+## Migration
+If the way a tag was calculated has changed, or new tags are added (or removed) or this feature was turned on for the first time (or turned on after being off), you must do the following: 
+- Manually run `trigger_update_object_tags` scheduled task from the UI, which queues a `update_object_tags` adhoc task that will process all objects marked as needing sync (default is true)
+or
+- Call the CLI to execute a `update_object_tags` adhoc task manually.
+
+## Reporting
+There is an additional graph added to the object summary report showing the tag value combinations and counts of each.
+
+Note, this is only for files that have been uploaded from this environment, and may not be consistent for environments where `overwriteobjecttags` is disabled (because the site does not know if a file was overwritten in the external store by another client).
+
+## For developers
+
+### Adding a new source
+Note the rules about sources:
+- Identifier must be < 32 chars long.
+- Value must be < 128 chars long.
+
+While external providers allow longer key/values, we intentionally limit it to reserve space for future use. These limits may change in the future as the feature matures.
+
+To add a new source:
+- Implement `tag_source`
+- Add to the `tag_manager` class
+- As part of an upgrade step, mark all objects `tagsyncstatus` to needing sync (using `tag_manager` class, or manually in the DB)
```

Comment:
I kinda want to lean towards not implementing methods unless there is an actual for them, since currently such a method would not be called anywhere.

I don't think at this stage manually updating the DB is that difficult, especially since it is explained in the readme what to do.

Reply:
Completely understandable. Part of the reason I brought this up was to improve the visibility since I missed reading it in the first place, but that'd be entirely on the user for not reading the instructions properly.